bgp/tests/local_tests/attributes_test.go

```diff
@@ -0,0 +1,217 @@
+					ReachPrefix: routeList[3],
+				},
+				ExpectedResultBeforePolicy: valpb.RouteTestResult_ROUTE_TEST_RESULT_ACCEPT,
+				ExpectedResult:             valpb.RouteTestResult_ROUTE_TEST_RESULT_NOT_PREFERRED,
+			}, {
+				Description: "Unpreferred route due to MED",
+				Input: &valpb.TestRoute{
+					ReachPrefix: routeList[4],
+				},
+				ExpectedResultBeforePolicy: valpb.RouteTestResult_ROUTE_TEST_RESULT_ACCEPT,
+				ExpectedResult:             valpb.RouteTestResult_ROUTE_TEST_RESULT_NOT_PREFERRED,
+			}},
+			LongerPathRouteTests: []*valpb.RouteTestCase{{
+				Description: "Accepted route due to higher local-pref",
+				Input: &valpb.TestRoute{
+					ReachPrefix: routeList[3],
+				},
+				ExpectedResultBeforePolicy: valpb.RouteTestResult_ROUTE_TEST_RESULT_NOT_PREFERRED,
+				ExpectedResult:             valpb.RouteTestResult_ROUTE_TEST_RESULT_ACCEPT,
+			}},
+			AlternatePathRouteTests: []*valpb.RouteTestCase{{
+				Description: "Accepted route due to lower MED",
+				Input: &valpb.TestRoute{
+					ReachPrefix: routeList[4],
+				},
+				ExpectedResultBeforePolicy: valpb.RouteTestResult_ROUTE_TEST_RESULT_NOT_PREFERRED,
+				ExpectedResult:             valpb.RouteTestResult_ROUTE_TEST_RESULT_ACCEPT,
+			}},
+		},
+		installPolicies: func(t *testing.T, dut1, dut2, dut3, dut4, dut5 *Device) {
+			if debug {
+				fmt.Println("Installing test policies")
+			}
+
+			dut1PolicyName := "set-attributes-dut1"
+			dut1Policy := &oc.RoutingPolicy_PolicyDefinition_Statement_OrderedMap{}
+			dut5PolicyName := "set-attributes-dut5"
+			dut5Policy := &oc.RoutingPolicy_PolicyDefinition_Statement_OrderedMap{}
+			rejectPolicyName := "export-policy"
+			rejectPolicy := &oc.RoutingPolicy_PolicyDefinition_Statement_OrderedMap{}
+
+			for i, route := range routeList {
+				// Create prefix set
+				prefixSetName := "accept-" + route
+				prefixPath := ocpath.Root().RoutingPolicy().DefinedSets().PrefixSet(prefixSetName).Prefix(route, "exact").IpPrefix()
+				Replace(t, dut1, prefixPath.Config(), route)
+				Replace(t, dut5, prefixPath.Config(), route)
+
+				var installDut1Stmt bool
+				dut1Stmt := &oc.RoutingPolicy_PolicyDefinition_Statement{Name: ygot.String(route + "-setattr-policy-dut1")}
+				dut1Stmt.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetPrefixSet(prefixSetName)
+				dut1Stmt.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+
+				var installDut5Stmt bool
+				dut5Stmt := &oc.RoutingPolicy_PolicyDefinition_Statement{Name: ygot.String(route + "-setattr-policy-dut5")}
+				dut5Stmt.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetPrefixSet(prefixSetName)
+				dut5Stmt.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+
+				var installRejectStmt bool
+				rejectStmt := &oc.RoutingPolicy_PolicyDefinition_Statement{Name: ygot.String(route + "-reject-policy")}
+
+				switch i {
```

Comment:
this isn't very clear. i don't see why i == 1 means set communites

Reply:
Added a comment. Each index refers to an input test route, and each one has its corresponding policies that applies to it. Each one is essentially a subtest that re-uses the topology.